### PR TITLE
add logout feature

### PR DESCRIPTION
**Pull request by MichelleJY**

### CODE REVIEW

**Comment by MichelleJY** [comment]
image
image

File: src/main/java/codeu/controller/LogoutServlet.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.mindrot.jbcrypt.BCrypt;
+
+/** Servlet class responsible for the logout page. */
+public class LogoutServlet extends HttpServlet {
+
+  /** Store class that gives access to Users. */
+  private UserStore userStore;
```
Comment (kyrabusser) [inline src/main/java/codeu/controller/LogoutServlet.java]: This UserStore isn't used, so you can remove it and all the references to it.

File: src/test/java/codeu/controller/LogoutServletTest.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    logoutServlet.doGet(mockRequest, mockResponse);
+    Mockito.verify(mockRequestDispatcher).forward(mockRequest, mockResponse);
+  }
+
+
+  @Test
+  public void testDoPost() throws IOException,ServletException{
+
+    HttpSession mockSession = Mockito.mock(HttpSession.class);
+    Mockito.when(mockRequest.getSession()).thenReturn(mockSession);
```
Comment (kyrabusser) [inline src/test/java/codeu/controller/LogoutServletTest.java]: I think it's a good idea for this test to also check that the doPost() method calls invalidate() on the session by using Mockito.verify().

File: src/test/java/codeu/controller/LogoutServletTest.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+    logoutServlet.doPost(mockRequest, mockResponse);
+    Mockito.verify(mockSession).invalidate();
```
Comment (MichelleJY) [inline src/test/java/codeu/controller/LogoutServletTest.java]: @kyrabusser Added this line. Is that right for verifying invalidate?

**Pull request merged**